### PR TITLE
Regnet weights v2

### PR DESCRIPTION
**Pull request by kazhang**
Better weights are obtained with the new recipe.

cc @datumbox @bjuncek

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -219,20 +219,20 @@ EfficientNet-B4                   83.384          96.594
...
+regnet_x_8gf                      81.682          95.678
+regnet_x_16gf                     82.716          96.196
+regnet_x_32gf                     83.014          96.288
+regnet_y_400mf                    75.804          92.742
+regnet_y_800mf                    78.828          94.502
+regnet_y_1_6gf                    80.876          95.444
+regnet_y_3_2gf                    81.982          95.972
+regnet_y_8gf                      82.828          96.330
+regnet_y_16gf                     82.886          96.328
+regnet_y_32gf                     83.368          96.498
```
Comment (datumbox) [inline docs/source/models.rst:235]: @kazhang This is a non-blocking comment for FYI and discussion.

We haven't updated these numbers for any other model that we refreshed. The reason is that the overall documentation of this section seems to imply that these accuracies are achieved by using the `pretrained=True` API. Since the old API gives you access to the old weights, we didn't change these numbers. This is debatable. Perhaps we could change the phrasing and update it. Or perhaps we should keep the numbers as-is for now. Thoughts?

The plan is to revamp the documentation of models in H1 with @NicolasHug and programmatically build the docs from the meta-data stored in the weights.

File: torchvision/models/regnet.py [hunk 2]
```diff
@@ -37,20 +37,20 @@
...
+    "regnet_y_8gf": "https://download.pytorch.org/models/regnet_y_8gf-dc2b1b54.pth",
+    "regnet_y_16gf": "https://download.pytorch.org/models/regnet_y_16gf-3e4a00f9.pth",
+    "regnet_y_32gf": "https://download.pytorch.org/models/regnet_y_32gf-8db6d4b5.pth",
+    "regnet_x_400mf": "https://download.pytorch.org/models/regnet_x_400mf-62229a5f.pth",
+    "regnet_x_800mf": "https://download.pytorch.org/models/regnet_x_800mf-94a99ebd.pth",
+    "regnet_x_1_6gf": "https://download.pytorch.org/models/regnet_x_1_6gf-a12f2b72.pth",
+    "regnet_x_3_2gf": "https://download.pytorch.org/models/regnet_x_3_2gf-7071aa85.pth",
+    "regnet_x_8gf": "https://download.pytorch.org/models/regnet_x_8gf-2b70d774.pth",
+    "regnet_x_16gf": "https://download.pytorch.org/models/regnet_x_16gf-ba3796d7.pth",
+    "regnet_x_32gf": "https://download.pytorch.org/models/regnet_x_32gf-6eb8fdc6.pth",
```
Comment (datumbox) [inline torchvision/models/regnet.py:53]: We shouldn't update these URLs. Instead we should deploy the weights on the new API. Perhaps you do this to easier verify the accuracy of the models using the existing scripts. If that's the case, ignore my comment. :)

**Review by datumbox — COMMENTED** [review]
It's likely I'm jumping uninvited to review a draft PR, apologies for snooping around. Just wanted actually to get early your input on how to document things, let me know what you think.

File: torchvision/prototype/models/regnet.py [hunk 3]
```diff
@@ -75,7 +75,17 @@ class RegNet_Y_400MF_Weights(WeightsEnum):
...
         },
     )
-    default = ImageNet1K_V1
+    ImageNet1K_V2 = Weights(
+        url="https://download.pytorch.org/models/regnet_y_400mf-e6988f5f.pth",
+        transforms=partial(ImageNetEval, crop_size=224, resize_size=232),
+        meta={
+            **_COMMON_META,
+            "recipe": "https://github.com/pytorch/vision/issues/3995#new-recipe",
+            "acc@1": 75.804,
```
Comment (datumbox) [inline torchvision/prototype/models/regnet.py:84]: Could you confirm that all validation metrics are estimated with batch-size 1?

**Review by datumbox — COMMENTED** [review]
LGTM, thanks!

**Review by datumbox — APPROVED** [review]

File: torchvision/prototype/models/regnet.py (hunk 3 above)
Comment (kazhang) [reply torchvision/prototype/models/regnet.py:84]: Yes, verified all the weights with the old API:
```bash
cd /data/home/kaizh/repos/vision/references/classification
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_400mf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_400mf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_800mf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_800mf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_1_6gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_1_6gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_3_2gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_3_2gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_8gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_8gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_16gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_16gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_x_32gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_x_32gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_400mf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_400mf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_800mf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_800mf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_1_6gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_1_6gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_3_2gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_3_2gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_8gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_8gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_16gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_16gf.out | grep "Test:  Acc@1"
srun -p dev --cpus-per-task=16 -t 5:00:00 --gpus-per-node=2 torchrun --nproc_per_node=1 train.py --model regnet_y_32gf --pretrained --test-only --val-resize-size 232 -b 1 | tee ~/logs/verify_regnet_y_32gf.out | grep "Test:  Acc@1"
```

Comment (kazhang) [reply torchvision/prototype/models/regnet.py:84]:
```bash
tail -n 2 ~/logs/verify_regnet_*
==> /data/home/kaizh/logs/verify_regnet_x_16gf.out <==
Test:  Total time: 0:21:04
Test:  Acc@1 82.716 Acc@5 96.196

==> /data/home/kaizh/logs/verify_regnet_x_1_6gf.out <==
Test:  Total time: 0:10:34
Test:  Acc@1 79.668 Acc@5 94.922

==> /data/home/kaizh/logs/verify_regnet_x_32gf.out <==
Test:  Total time: 0:21:04
Test:  Acc@1 83.014 Acc@5 96.288

==> /data/home/kaizh/logs/verify_regnet_x_3_2gf.out <==
Test:  Total time: 0:18:13
Test:  Acc@1 81.196 Acc@5 95.430

==> /data/home/kaizh/logs/verify_regnet_x_400mf.out <==
Test:  Total time: 0:11:31
Test:  Acc@1 74.864 Acc@5 92.322

==> /data/home/kaizh/logs/verify_regnet_x_800mf.out <==
Test:  Total time: 0:08:57
Test:  Acc@1 77.522 Acc@5 93.826

==> /data/home/kaizh/logs/verify_regnet_x_8gf.out <==
Test:  Total time: 0:20:10
Test:  Acc@1 81.682 Acc@5 95.678

==> /data/home/kaizh/logs/verify_regnet_y_16gf.out <==
Test:  Total time: 0:22:47
Test:  Acc@1 82.886 Acc@5 96.328

==> /data/home/kaizh/logs/verify_regnet_y_1_6gf.out <==
Test:  Total time: 0:22:04
Test:  Acc@1 80.876 Acc@5 95.444

==> /data/home/kaizh/logs/verify_regnet_y_32gf.out <==
Test:  Total time: 0:20:00
Test:  Acc@1 83.368 Acc@5 96.498

==> /data/home/kaizh/logs/verify_regnet_y_3_2gf.out <==
Test:  Total time: 0:17:23
Test:  Acc@1 81.982 Acc@5 95.972

==> /data/home/kaizh/logs/verify_regnet_y_400mf.out <==
Test:  Total time: 0:12:47
Test:  Acc@1 75.804 Acc@5 92.742

==> /data/home/kaizh/logs/verify_regnet_y_800mf.out <==
Test:  Total time: 0:11:37
Test:  Acc@1 78.828 Acc@5 94.502

==> /data/home/kaizh/logs/verify_regnet_y_8gf.out <==
Test:  Total time: 0:17:12
Test:  Acc@1 82.828 Acc@5 96.330
```

**Comment by kaijieshi7** [comment]
Why haven't updated weightv2's link in regnet.py?

**Comment by datumbox** [comment]
@kaijieshi7 For Backwards compatibility reasons. Soon we will rollout the new multi-weight support  API on main area and you will be able to access them easier
